src/connector/src/sink/iceberg/mod.rs

```diff
@@ -1378,6 +1573,36 @@ impl SinkCommitCoordinator for IcebergSinkCommitter {
             .partition_type(schema)
             .map_err(|err| SinkError::Iceberg(anyhow!(err)))?;
 
+        let txn = Transaction::new(&self.table);
+        let snapshot_id = txn.generate_unique_snapshot_id();
```

Comment:
Only generate snapshot id when `is_normal_commit == true`, otherwise, we should use `snapshot_id` recorded in the system table.